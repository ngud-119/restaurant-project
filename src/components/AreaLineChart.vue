<template>
  <div class="line-chart">
    <div class="details">
      <div>
        <v-icon class="icon" size="32" :icon="data.icon"></v-icon>
      </div>
      <div class="content">
        <p>{{ data.name }}</p>
        <h4>{{ data.stat }}</h4>
      </div>
    </div>
    <div class="chart">
      <Line :data="data.chartData" :options="data.options" />
    </div>
  </div>
</template>

<script>
import {
  Chart as ChartJS,
  CategoryScale,
  LinearScale,
  PointElement,
  LineElement,
  Title,
  Tooltip,
  Legend,
  Filler
} from 'chart.js'

import { Line } from 'vue-chartjs'

ChartJS.register(
  CategoryScale,
  LinearScale,
  PointElement,
  LineElement,
  Title,
  Tooltip,
  Legend,
  Filler
)

export default {
  name: 'areaLineChart',
  props: ['data'],
  components: {
    // eslint-disable-next-line vue/no-reserved-component-names
    Line
  },
  data() {
    return {}
  }
}
</script>

<style lang="scss" scoped>
@import '../assets/config';
@import '../assets/responsive';
.line-chart {
  background-color: rgba(234, 234, 234, 0.549);
  background-color: white;
  backdrop-filter: blur(10px);
  border-radius: 10px;
  .details {
    display: flex;
    align-items: center;
    gap: 14px;
    padding: 20px;
    .icon {
      color: $primary;
      background-color: rgba(204, 8, 11, 0.2);
      padding: 26px;
      border-radius: 100%;
    }

    .content {
      p {
        font-size: 14px;
        color: rgb(130, 130, 130);
        text-transform: capitalize;
      }
      h4 {
        font-size: 28px;
      }
    }
  }

  .chart {
    padding: 0px 0px;
  }
}
</style>
