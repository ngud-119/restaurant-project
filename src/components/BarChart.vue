<template>
  <div class="bar-chart">
    <div class="details d-flex justify-space-between">
      <h4 class="font-weight-bold">Sales Figures</h4>
      <div>
        <v-select
          :hide-details="true"
          class="year-filter"
          :items="barYearFilter"
          item-title="value"
          item-value="key"
          v-model="selectedYear"
          variant="outlined"
          color="#79a33d"
          density="compact"
        ></v-select>
      </div>
    </div>
    <div class="chart">
      <Bar style="height: 300px" :data="barData" :options="barChartOptions" />
    </div>
  </div>
</template>

<script>
import { Bar } from 'vue-chartjs'
import { barData, barChartOptions } from './graph-data/barChart'
import {
  Chart as ChartJS,
  Title,
  Tooltip,
  Legend,
  BarElement,
  CategoryScale,
  LinearScale
} from 'chart.js'

ChartJS.register(CategoryScale, LinearScale, BarElement, Title, Tooltip, Legend)

export default {
  name: 'barChart',
  components: {
    Bar
  },
  data() {
    return {
      barData: barData,
      barChartOptions: barChartOptions,
      selectedYear: { key: 0, value: '2023 - 2024' },
      barYearFilter: [
        { key: 0, value: '2023 - 2024' },
        { key: 1, value: '2022 - 2023' },
        { key: 2, value: '2021 - 2022' }
      ]
    }
  }
}
</script>

<style lang="scss" scoped>
.bar-chart {
  background-color: rgba(234, 234, 234, 0.549);
  background-color: white;
  backdrop-filter: blur(10px);
  border-radius: 10px;
  padding: 20px;

  .details {
    margin-bottom: 20px;

    h4 {
      font-size: 28px;
    }

    .year-filter {
      min-width: 140px;
    }
  }
}
</style>
7654    