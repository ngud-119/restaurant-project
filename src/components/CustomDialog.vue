<template>
  <div class="text-center">
    <v-btn color="primary">
      Open Dialog
      <v-dialog  v-model="dialog" activator="parent" width="auto">
        <v-card>
          <v-card-item>
            <div class="text-center text-h3">
              <v-icon color="primary" :icon="icon"></v-icon>
            </div>
            <div>
              <h4 class="text-h4">{{ heading }}</h4>
            </div>
            <v-card-text class="text-center">
              {{ text }}
            </v-card-text>
          </v-card-item>
          <v-card-actions>
            <v-spacer></v-spacer>
            <v-btn color="green-darken-1" variant="text" @click="dialog = false"> Disagree </v-btn>
            <v-btn color="green-darken-1" variant="text" @click="dialog = false"> Agree </v-btn>
          </v-card-actions>
        </v-card>
      </v-dialog>
    </v-btn>
  </div>
</template>

<script>
export default {
  name: 'CustomDialog',
  props: {
    icon: String,
    heading: String,
    text: String
  },
  data() {
    return {
      dialog: false
    }
  }
}
</script>

<style lang="scss" scoped></style>
