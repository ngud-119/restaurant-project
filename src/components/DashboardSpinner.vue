<template>
    <div class="loading-container">
        <div class="loader">
            <swapping-squares-spinner :animation-duration="1000" :size="80" color="#cc080b" />
        </div>
    </div>
</template>

<script>
import { SwappingSquaresSpinner } from 'epic-spinners'
export default {
    name: "DashboardSpinner",
    components: {
        SwappingSquaresSpinner
    },
}
</script>

<style lang="scss" scoped>
.loading-container {
    position: absolute;
    background-color: rgba(255, 255, 255, 0.6);
    width: 100vw;
    height: 90vh;
    z-index: 10;

    .loader {
        height: 100%;
        transform: translate(40%, 45%);
    }
}
</style>