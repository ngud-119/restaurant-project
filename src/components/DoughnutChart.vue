<template>
  <div class="doughnut-chart">
    <div class="details d-flex justify-space-between">
      <div>
        <h4 class="font-weight-bold">Popular Food</h4>
        <span>On this week</span>
      </div>
      <div>
        <v-select
          :hide-details="true"
          class="year-filter"
          :items="barYearFilter"
          item-title="value"
          item-value="key"
          v-model="selectedYear"
          variant="outlined"
          color="#79a33d"
          density="compact"
        ></v-select>
      </div>
    </div>
    <div class="chart">
      <Doughnut :data="data.doughnutData" :options="data.doughnutOptions" />
    </div>
  </div>
</template>

<script>
import { Chart as ChartJS, ArcElement, Tooltip, Legend } from 'chart.js'
import { Doughnut } from 'vue-chartjs'
ChartJS.register(ArcElement, Tooltip, Legend)

export default {
  name: 'doughnutChart',
  props: ['data'],
  components: {
    Doughnut
  },
  data() {
    return {
      selectedYear: { key: 0, value: '01 - 07, Jan' },
      barYearFilter: [
        { key: 0, value: '01 - 07, Jan 2024' },
        { key: 1, value: '08 - 14, Feb 2024' },
        { key: 2, value: '15 - 21, Mar 2024' }
      ]
    }
  }
}
</script>

<style lang="scss" scoped>
@import '../assets/config';
@import '../assets/responsive';
.doughnut-chart {
  background-color: rgba(234, 234, 234, 0.549);
  background-color: white;
  backdrop-filter: blur(10px);
  border-radius: 10px;
  padding: 20px;

  .details {
    margin-bottom: 20px;

    h4 {
      font-size: 22px;
    }
    span {
      font-size: 14px;
      color: rgb(130, 130, 130);
    }
  }
}
</style>
