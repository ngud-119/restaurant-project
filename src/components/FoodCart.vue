<template>
  <v-navigation-drawer
    class="cart-drawer"
    v-model="$store.state.cartVisibility"
    temporary
    location="right"
  >
    <div class="cart-container">
      <div class="header pa-4 d-flex align-center justify-space-between">
        <h2>Cart</h2>
        <v-icon icon="mdi-close-circle-outline" @click.stop="toggleCart"></v-icon>
      </div>
      <div class="cart-item-container">
        <div v-for="(item, index) in getMyCart.items" :key="index" class="cart-item">
          <div class="d-flex align-center justify-space-between pa-3">
            <div class="d-flex align-center">
              <div class="cart-image-container me-5 d-flex align-center">
                <img
                  class="rounded-circle"
                  :src="imageUrl + 'food/' + item.foodImage"
                  :alt="item.name"
                  width="70"
                  height="70"
                />
              </div>
              <div>
                <h4 class="food-name font-weight-bold mb-2">{{ item.foodName }}</h4>
                <div class="cart-control border">
                  <div class="cart-control-btn increase-btn border" @click="decreaseCart(item)">
                    <v-icon class="icon" icon="mdi-minus"></v-icon>
                  </div>
                  <div class="cart-control-text border">
                    <p>
                      {{ item.quantity }}
                    </p>
                  </div>
                  <div class="cart-control-btn decrease-btn border" @click="addToCart(item)">
                    <v-icon class="icon" icon="mdi-plus"></v-icon>
                  </div>
                </div>
              </div>
            </div>
            <div>
              <div class="text-right">
                <v-icon
                  class="remove-from-cart-icon d-inline-block"
                  icon="mdi-trash-can-outline"
                  @click="removeCart(item)"
                ></v-icon>
              </div>
              <div class="mt-5">
                <p>Price: {{ item.totalPrice }} ৳</p>
              </div>
            </div>
          </div>
          <v-divider thickness="2px" class=""></v-divider>
        </div>
      </div>
      <div class="cart-footer" v-if="getMyCart?.items?.length > 0">
        <div class="pa-4 mt d-flex justify-space-between subtotal">
          <p>Subtotal</p>
          <p>{{ getMyCart.amount }} ৳</p>
        </div>
        <v-divider thickness="2px"></v-divider>
        <div class="text-center pa-4 mt-2">
          <button class="confirm-btn" @click="confirmOrder()">Confirm Order</button>
        </div>
      </div>
      <div class="empty-cart-text" v-if="!getMyCart.items || getMyCart?.items?.length <= 0">
        <p>Cart is Empty</p>
      </div>
    </div>
  </v-navigation-drawer>
</template>

<script>
import ApiCall from '../api/apiInterface'
import { imageUrl } from '../constants/config'
import { Store, mapActions, mapGetters, storeKey } from 'vuex'
import { useToast } from 'vue-toast-notification'
const $toast = useToast()

export default {
  name: 'FoodCart',
  data() {
    return {
      imageUrl: imageUrl
    }
  },
  watch: {},
  computed: {
    ...mapGetters({
      getCurrentUser: 'getCurrentUser',
      getDrawerVisibility: 'getDrawerVisibility',
      getLoadingState: 'getLoadingState',
      getMyCart: 'getMyCart'
    })
  },
  methods: {
    ...mapActions({
      logoutUser: 'logoutUser',
      toggleCartVisibility: 'toggleCartVisibility',
      updateCart: 'updateCart',
      decreaseCartItem: 'decreaseCartItem',
      increaseCartItem: 'increaseCartItem',
      removeCartItem: 'removeCartItem',
      setTotalAmount: 'setTotalAmount'
    }),
    toggleCart() {
      this.toggleCartVisibility()
    },
    addToCart(food) {
      this.increaseCartItem(food)
      this.setTotalAmount()
    },

    decreaseCart(food) {
      this.decreaseCartItem(food)
      this.setTotalAmount()
    },

    removeCart(food) {
      this.removeCartItem(food)
      this.setTotalAmount()
    },

    async confirmOrder() {
      //const cartDetails = { ...this.getMyCart }
      // const currentDate = new Date()
      // const isoString = currentDate.toISOString()
      try {
        const cartDetails = JSON.parse(JSON.stringify(this.getMyCart))

        cartDetails.items.map((item) => {
          delete item.foodImage
          delete item.foodName
          item.foodPackageId = null
        })
        cartDetails.orderNumber = new Date().toISOString()
        cartDetails.phoneNumber = null
        //cartDetails.orderStatus = 1

        const response = await ApiCall.post('api/order/create', cartDetails)
        console.log(response)
        $toast.success('Order created successfully')
        this.toggleCartVisibility()
        this.updateCart({
          tableId: 0,
          orderNumber: '',
          amount: 0,
          phoneNumber: '',
          items: []
        })
      } catch (error) {
        console.log(error)
      }
    }
  }
}
</script>

<style lang="scss" scoped>
@import '../assets/config';
@import '../assets/responsive';

.cart-drawer {
  .cart-container {
    .header {
      color: white;
      background-color: $primary;
      position: sticky;
      top: 0;
      left: 0;
      z-index: 2;
    }

    .cart-footer {
      position: sticky;
      bottom: 0;
      background-color: #ffffff;
      .subtotal {
        font-weight: bold;
        font-size: 16px;
      }
      .confirm-btn {
        display: block;
        width: 100%;
        color: white;
        font-size: 16px;
        background-color: $primary;
        padding: 15px 0px;

        &:hover {
          background-color: darken($primary, 10);
        }
      }
    }

    .empty-cart-text {
      height: 90vh;
      display: flex;
      justify-content: center;
      align-items: center;

      p {
        font-size: 24px;
      }
    }

    .cart-item-container {
      .cart-item {
        .food-name {
          font-size: 18px;
        }
        .cart-control {
          display: inline-flex;
          height: fit-content;

          .cart-control-btn {
            font-size: 20px;
            cursor: pointer;
            padding: 0px 10px;

            &:hover {
              color: white;
              background-color: $primary;
            }
            .icon {
              font-size: 14px;
            }
          }

          .cart-control-text {
            display: flex;
            align-items: center;
            justify-content: center;
            width: 50px;
            p {
              font-size: 16px;
            }
          }
        }

        .remove-from-cart-icon {
          color: darken(#ffffff, $amount: 50);
          &:hover {
            cursor: pointer;
            color: $primary;
          }
        }
      }
    }
  }
}
</style>
