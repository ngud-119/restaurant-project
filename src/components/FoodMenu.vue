<template>
	<div>

	</div>
</template>

<script>
export default {
	name: "FoodMenu",
}
</script>

<style lang="scss" scoped>
@import "../assets/config";
@import "../assets/responsive";
</style>