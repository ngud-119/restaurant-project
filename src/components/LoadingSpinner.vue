<template>
	<div class="loading-container">
		<swapping-squares-spinner :animation-duration="1000" :size="80" color="#cc080b" />
	</div>
</template>

<script>
import { SwappingSquaresSpinner } from 'epic-spinners'
export default {
	name: "LoadingSpinner",
	components: {
		SwappingSquaresSpinner
	},
}
</script>

<style lang="scss" scoped>
.loading-container {
	position: absolute;
	background-color: rgba(255, 255, 255, 0.6);
	width: 100vw;
	height: 100vh;
	display: flex;
	justify-content: center;
	align-items: center;

}
</style>