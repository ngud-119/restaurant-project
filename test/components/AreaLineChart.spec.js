import { expect, test, describe } from 'vitest'
import { mount, shallowMount } from '@vue/test-utils'
import AreaLineChart from '../../src/components/AreaLineChart.vue'
import { createVuetify } from 'vuetify'
import * as components from 'vuetify/components'
import * as directives from 'vuetify/directives'
const vuetify = createVuetify({
  components,
  directives
})

//!-------- snapshot test ---------
// test('mount vue component', () => {
//   const wrapper = mount(AreaLineChart)
//   expect(wrapper.html()).toMatchSnapshot()
// })

//!-------- checking props of a component ---------
describe('props__check', () => {
  test('props__data__check', () => {
    const data = { name: 'Income' }
    const wrapper = shallowMount(AreaLineChart, {
      props: { data },
      global: {
        stubs: {
          Line: {
            template: '<canvas />'
          }
        },
        plugins: [vuetify]
      }
    })

    expect(wrapper.text()).toContain(data.name)
  })

  test('sanity test', () => {
    expect(true).toBe(true)
  })
})

//!-------- test a tag/element is exist or not ---------
describe('test__tag', () => {
  const data = { name: 'Income' }
  const wrapper = mount(AreaLineChart, {
    props: { data },
    global: {
      stubs: {
        Line: {
          template: '<canvas />'
        }
      }
    }
  })

  const paraElement = wrapper.find('p')
  const vIcon = wrapper.find('v-icon')

  test('is__contain__paragraph__tag', () => {
    expect(paraElement.exists()).toBe(true)
  })

  test('is__contain__props__data__name', () => {
    expect(paraElement.text()).toContain(data.name)
  })

  test('is__v-icon__contains__icon__props', () => {
    expect(vIcon.exists()).toBe(true)
  })
})
