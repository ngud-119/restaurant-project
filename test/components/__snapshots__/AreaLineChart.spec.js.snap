// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`mount vue component 1`] = `
"<div data-v-f3b1501a="" class="line-chart">
  <div data-v-f3b1501a="" class="details">
    <div data-v-f3b1501a="">
      <!-- <v-icon class="icon" size="32" :icon="data.icon"></v-icon> -->
    </div>
    <div data-v-f3b1501a="" class="content">
      <!-- <p>{{ data.name }}</p>
        <h4>{{ data.stat }}</h4> -->
    </div>
  </div>
  <div data-v-f3b1501a="" class="chart">
    <!-- <Line :data="data.chartData" :options="data.options" /> -->
  </div>
</div>"
`;
